client/data/geo/test/use-geolocation-query.test.tsx

```diff
@@ -0,0 +1,58 @@
+/**
+ * @jest-environment jsdom
+ */
+
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
+import { renderHook, waitFor } from '@testing-library/react';
+import { when } from 'jest-when';
+import { act, ReactNode } from 'react';
+import { useGeoLocationQuery, GeoLocationData } from '../use-geolocation-query';
+
+jest.useFakeTimers();
+
+describe( 'useGeoLocationQuery', () => {
+	const wrapper = ( { children }: { children: ReactNode } ) => (
+		<QueryClientProvider client={ new QueryClient() }>{ children }</QueryClientProvider>
+	);
+
+	const mockGeoData: GeoLocationData = {
+		city: 'San Francisco',
+		country_long: 'United States',
+		country_short: 'US',
+		latitude: '37.7749',
+		longitude: '-122.4194',
+		region: 'California',
+	};
+
+	beforeAll( () => {
+		when( jest.spyOn( globalThis, 'fetch' ) )
+			.calledWith( 'https://public-api.wordpress.com/geo/' )
+			.mockResolvedValue( { json: () => Promise.resolve( mockGeoData ) } as Response );
+	} );
+
+	it( 'should fetch geolocation data', async () => {
+		const { result } = renderHook( () => useGeoLocationQuery(), { wrapper } );
+
+		// Initially loading
+		expect( result.current.isLoading ).toBe( true );
+
+		// Wait for the query to complete
+		await waitFor( () => expect( result.current.isSuccess ).toBe( true ) );
+
+		// Verify data
+		expect( result.current.data ).toEqual( mockGeoData );
+	} );
+
+	it( 'should not refetch', async () => {
+		const { result } = renderHook( () => useGeoLocationQuery(), { wrapper } );
+
+		// Wait for the query to complete
+		await waitFor( () => expect( result.current.isSuccess ).toBe( true ) );
+
+		act( () => {
+			result.current.refetch();
+		} );
+
+		expect( result.current.isRefetching ).toBe( false );
```

Comment:
Isn't it a better way to just rerender the hook? Something like this:

```suggestion
		const { rerender, result } = renderHook( () => useGeoLocationQuery(), { wrapper } );

		// Wait for the query to complete
		await waitFor( () => expect( result.current.isSuccess ).toBe( true ) );

		rerender();

		expect( result.current.isRefetching ).toBe( false );
```

It would also help get rid of the odd `act()` call.

Reply:
I think I had tried this in an earlier iteration, and while it _does_ pass, it also passes on `trunk`, so not actually testing the desired behavior. That being said, I'm seeing the same with the current test implementation, so I'll have to take a look and see why that is.

Reply:
Addressed this as part of 5732f1947bb81e50fd855a00d8ba25041712f302. Ideally we'd also be testing the persistence behavior, but I found that to be very clunky to do without stubbing out a lot of the persistence behavior through the React Query provider. I think it's reasonable enough to just have a basic test that it doesn't immediately become stale 🤷